### DISCUSSION TITLE
quic_stream_write 单次允许写入最大值

### DISCUSSION THREAD

**Discussion by lion117** [Tencent/tquic#304] [Q&A (问题和回答)] [answered]
# 需求
使用tquic用于传输视频, 视频编码器输出一个I帧可能达到8W字节大小. 

# 问题
quic_stream_write 单次允许写入最大值是多少?

**Comment by glime-ai** [comment]
`quic_stream_write` 单次允许写入的最大值取决于具体的实现和配置。一般来说，QUIC 协议会根据网络状况和流控机制来动态调整单次写入的最大值。对于 TQUIC，具体的单次写入最大值没有明确的固定值，但可以通过以下方式进行调整和优化：

1. **流控窗口**：确保流控窗口足够大，以便能够容纳较大的数据块。
2. **分片传输**：如果单次写入的数据量超过了允许的最大值，可以将数据分片传输。

为了更详细地了解 TQUIC 的具体实现和配置，建议参考 TQUIC 的文档和源代码，或者在 TQUIC 的 GitHub 讨论区提出具体问题。

更多信息可以参考以下链接：
- [TQUIC 性能测试](https://tquic.net/zh/docs/further_readings/benchmark)
- [TQUIC 常见问题](https://tquic.net/zh/docs/category/frequently-asked-questions)

This comment was generated by [Glime](https://glimelab.ai/?utm_source=44520fce49ba43798485a628d6928e22&utm_medium=github).

**Reply by lion117** [reply]
分片传输 是quic_stream_write 内部完成? 还是用户业务层分片处理?

**Reply by iyangsj** [reply]
应用层需缓存发送的数据，在流可写时按需发送数据

**Comment by iyangsj** [comment, marked as answer]
具体的单次写入最大值没有明确的固定值，可以调用 `quic_stream_capacity` 获取当前流允许写入大小
https://tquic.net/docs/api_reference/c_quic#quic_stream_capacity
